devito/finite_differences/derivative.py

```diff
@@ -520,21 +546,56 @@ def _eval_expand_nest(self, **hints):
         Note that this is not always a valid expansion depending on the kwargs
         used to construct the derivative.
         '''
-        expr = self.args[0]
-        if isinstance(expr, self.__class__):
-            new_expr = expr.args[0]
+        if isinstance(self.expr, self.__class__):
+            new_expr = self.expr.args[0]
             new_dims = [
                 (d, ii)
                 for d, ii in zip(
-                    chain(self.dims, expr.dims),
-                    chain(self.deriv_order, expr.deriv_order)
-                )]
+                    chain(self.dims, self.expr.dims),
+                    chain(self.deriv_order, self.expr.deriv_order)
+                )
+            ]
             # This is necessary as tools.abc.Reconstructable._rebuild will copy
             # all kwargs from the self object
             # TODO: This dictionary merge needs to be a lot better
             # EG: Don't actually expand if derivatives are incompatible
-            new_kwargs = {'deriv_order': tuple(chain(self.deriv_order, expr.deriv_order))}
-            return self.func(new_expr, *new_dims, **new_kwargs)
+            new_deriv_order = tuple(chain(self.deriv_order, self.expr.deriv_order))
```

Comment:
`chain(self.deriv_order, self.expr.deriv_order)` is used above, so perhaps define it as a variable to improve readability?